lib/link_checker.rb

```diff
@@ -0,0 +1,46 @@
+require 'yaml'
+require 'rest-client'
+require_relative 'root'
+require_relative 'app_config'
+require_relative 'url_checker'
+
+
+
+class LinkChecker
+  def initialize
+    @urls = YAML.load_file(GOVUK_PAGES_FILE)
```

Comment:
I'm not sold on there being a readability benefit from using `attr_accessor` etc; it looks a bit prettier, but removes a signal that distinguishes local variables from instance variables.

There is one concrete benefit, which is that if you make a typo in an access to an instance variable, ruby will silently return nil.  If you make a typo in a variable being read using `attr_*`, you'll get an error.  It's desirable to get the error early; but if you have sufficient testing this also won't be an issue.

In the absence of clear evidence, I wouldn't worry about this whichever style is used.